### ISSUE TITLE
Inconsistency in pulling provider-specific settings for SAML use case

### ISSUE THREAD

**Issue by agorka-contractsafe** [pennersr/django-allauth#3439]
When going through the SSO flow (for our SAML usage), the current code uses a mix of `provider` and `provider_id` to get settings. We use the `VERIFIED_EMAIL` setting in the `saml` provider settings, which works as expected. In `cleanup_email_addresses` where we get the settings from `settings = self.get_settings()` and `self.id` is `saml`:

```
def get_settings(self):
        return app_settings.PROVIDERS.get(self.id, {})

def cleanup_email_addresses(self, email, addresses, email_verified=False):
    ...
    settings = self.get_settings()
    verified_email = settings.get("VERIFIED_EMAIL", False)
    if verified_email:
      ...
```

But in the `SocialLogin` model, when we look for a provider's specific `EMAIL_AUTHENTICATION` setting, we do:

```
def lookup(self):
    """Look up the existing local user account to which this social login
    points, if any.
    """
    if not self._lookup_by_socialaccount():
        if app_settings.EMAIL_AUTHENTICATION or app_settings.PROVIDERS.get(
            self.account.provider, {}
        ).get("EMAIL_AUTHENTICATION", False):
            self._lookup_by_email()
```
	
In the provider's `sociallogin_from_response`, the provider gets set to `provider_id`, which in the `saml` case can be something like an IdP entity ID:

```
socialaccount = SocialAccount(
    extra_data=extra_data,
    uid=uid,
    provider=self.app.provider_id or self.app.provider,
)
```

We've pulled the `EMAIL_AUTHENTICATION` setting up to the root social account level (`SOCIALACCOUNT_EMAIL_AUTHENTICATION`), but wanted to report the inconsistency since I think we'd expect them to be able to pull from the provider-specific settings (vs individual IdP specific settings).

Let me know if you have any questions. Thanks!

cc: @ryanhiebert

**Comment by pennersr** [comment]
Fixed via be779dfe

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3439
